### PR TITLE
Prioritise primary stream

### PR DESCRIPTION
**Pull request by emhane**
programmability spans up a new attack surface onto the node. this pr prioritises the primary stream. it is a good precaution that aims to limit the damage of user-defined vulnerabilities to the injected handles, and not let them effect the whole node.

by letting users inject cap-handles into reth, we are making the node vulnerable to all kinds of errors we cannot foresee and therefore not make any guarantees for how the node runs. a scenario that could happen before, is that some user-defined cap-handle can be dos:ed so that it sends endless messages, this means that the `RlpxSatelliteStream` stream will get stuck in a loop and no more `eth` messages would be sent or received.

### CODE REVIEW

File: crates/net/eth-wire/src/multiplex.rs [hunk 1]
```diff
@@ -487,19 +488,17 @@ where
...
-                            break
-                        }
+                match proto.poll_next_unpin(cx) {
+                    Poll::Ready(Some(msg)) => {
+                        this.inner.out_buffer.push_back(msg);
+                    }
+                    Poll::Ready(None) => return Poll::Ready(None),
+                    Poll::Pending => {
+                        this.inner.protocols.push(proto);
+                        break
```
Comment (mattsse) [inline crates/net/eth-wire/src/multiplex.rs:501]: this break will terminate the for loop now, but we still want to advance the remaining streams, right?

perhaps we should rewrite `protocols: Vec<ProtocolStream>` as 
`FuturesUnordered<StreamFuture<ProtocolStream>`

see also SelectAll impl 

https://docs.rs/futures-util/0.3.30/src/futures_util/stream/select_all.rs.html#93

**Review by mattsse — CHANGES_REQUESTED** [review]
oh yeah this is better.

we're still draining all protocol messages but hitting the primary earlier again.

one comment about break statement and FuturesUnordered suggestion

File: crates/net/eth-wire/src/multiplex.rs (hunk 1 above)
Comment (emhane) [reply crates/net/eth-wire/src/multiplex.rs]: absolutely. I also found some more code that impedes us from prioritising the primary stream, fixed it in the latest commit.

**Comment by emhane** [comment]
stoping development, in favour of using `MuxDemuxStream` from https://github.com/paradigmxyz/reth/pull/5577 with `ConnectionHandler`.

**Pull request closed without merging**